src/main/java/com/flowpowered/engine/registry/BlockMaterialRegistry.java

```
@@ -0,0 +1,184 @@
+
+	public BlockMaterial getBlockMaterialByName(String blockMaterialName) {
+		checkIfRegistryIsInitialized();
+		checkIfBlockMaterialNameExists(blockMaterialName);
+		Integer id = yamlRegistry.getIdByName(blockMaterialName);
+		return getBlockMaterialByID(id);
+	}
+
+	public BlockMaterial getBlockMaterialByID(Integer id) {
+		checkIfRegistryIsInitialized();
+		checkIfBlockMaterialIdExists(id);
+		return idRegistryMap.get(id);
+	}
+
+	public boolean revertToLastCustomBlockBaseMaterial(Integer id) {
+		checkIfRegistryIsInitialized();
+		checkIfBlockMaterialIdExists(id);
+		BlockMaterial blockMaterial = idRegistryMap.get(id);
+		return blockMaterial.revertToLastCustomBlockBaseMaterial();
+	}
+
+	private BlockMaterial addBlockMaterialToRegistry(Integer id, BlockBaseMaterial blockBaseMaterial, BlockBaseMaterial customBlockBaseMaterial) {
+		checkIfRegistryIsInitialized();
+		String blockMaterialName = blockBaseMaterial.getName();
+		yamlRegistry.addIdName(id, blockMaterialName);
+		BlockMaterial blockMaterial = BlockMaterial.createBlockMaterial(blockBaseMaterial, customBlockBaseMaterial);
+		idRegistryMap.put(id, blockMaterial);
+		blockMaterial.setID(id);
+		return blockMaterial;
+	}
+
+	private boolean initializeRegistry() {
+		if (initialized) {
+			Flow.fine("This BlockRegistry is already initialized, doing nothing!");
+			return true;
+		}
+		if (Flow.getPlatform().isServer()) {
+			if (yamlRegistry.checkIfRegistryStorageExists(blockRegistryFileName)) {
+				if (!loadRegistry()) {
+					throw new RegistryIOException("BlockRegistry", "loaded");
+				}
+			}
+		}
+		initialized = true;
+		return initialized;
+	}
+
+	public boolean loadRegistry() {
+		if (Flow.getPlatform().isClient()) {
+			Flow.debug("Won't be loading registry on client platform!");
+			return true;
+		}
+		return yamlRegistry.load(blockRegistryFileName);
+	}
+
+	public boolean saveRegistry() {
+		if (Flow.getPlatform().isClient()) {
+			Flow.debug("Won't be saving registry on client platform!");
```

This should just throw an error. We don't want to even call this on the client.

Also, do !isServer, because Singleplayer is both.